shared/src/main/scala/coop/rchain/shared/Cell.scala

```diff
@@ -20,6 +21,21 @@ trait Cell[F[_], S] {
 object Cell {
   def apply[F[_], S](implicit ev: Cell[F, S]): Cell[F, S] = ev
 
+  def readerT[F[_], E, S](cell: Cell[F, S]): Cell[ReaderT[F, E, ?], S] =
+    new Cell[ReaderT[F, E, ?], S] {
+      override def modify(f: S => S): ReaderT[F, E, Unit] = ReaderT.liftF(cell.modify(f))
+
+      override def flatModify(f: S => ReaderT[F, E, S]): ReaderT[F, E, Unit] =
+        ReaderT { e =>
+          val af: S => F[S] = s => f(s).run(e)
```

Comment:
I'm a bit weary of the `.run` here - can't we avoid it somehow?
I'm asking b/c of performance and/or correctness. I'm also unsure how this affects stacksafety.
If this looks like PTSD - it is.